src/components/AbridgedSchedule.astro

```diff
@@ -0,0 +1,124 @@
+---
+import { getCollection, getEntry } from "astro:content";
+import { marked } from "marked";
+marked.use({ mangle: false, headerIds: false });
+import { localizeTime, formatTime } from "../utils";
+
+const exams = (await getCollection("exams"))
+  .map((entry) => {
+    return {
+      type: "exam",
+      title: entry.data.title,
+      date: localizeTime(entry.data.date),
+      readings: null,
+      content: entry.data.content,
+      notes: entry.data.notes,
+    };
+  })
+  .sort((a, b) => +a.date! - +b.date!);
+
+const homeworks = (
+  await getCollection("docs", (entry) => entry.id.startsWith("homework"))
+)
+  .flatMap((entry) => {
+    return [
+      {
+        slug: entry.slug,
+        type: "homework due",
+        title: entry.data.title,
+        date: localizeTime(entry.data.due),
+        readings: null,
+        content: null,
+        notes: null,
+      },
+    ];
+  })
+  .sort((a, b) => +a.date! - +b.date!);
+
+const lectures = (
+  await getCollection("docs", (entry) => entry.id.startsWith("lecture"))
+)
+  .map((entry) => {
+    return {
+      slug: entry.slug,
+      type: "lecture",
+      title: entry.data.title,
+      date: localizeTime(entry.data.date),
+      readings: entry.data.readings,
+      content: null,
+      notes: entry.data.notes,
+    };
+  })
+  .sort((a, b) => +a.date! - +b.date!);
+
+const now = new Date().getTime();
+
+// Filtering the sorted lists by what has a date later than today
+// and subsequently only taking the first in the list
+
+const nextHomework = homeworks.filter(function (x) {
+  return x.date.getTime() - now > 0;
+})[0];
+
+const nextLecture = lectures.filter(function (x) {
+  return x.date.getTime() - now > 0;
+})[0];
+
+const nextExam = exams.filter(function (x) {
+  return x.date.getTime() - now > 0;
+})[0];
+
+// Merging the three into one array
+const events = [nextHomework, nextLecture, nextExam];
```

Comment:
This code runs at build time, so it will be "incorrect" between when a due date happens and the next time that the web site gets updated. There's an SSR fix and a client side fix for this, but because this isn't a real PR but part of a class, I will not be explaining them (which would normally be kind of a jerk thing to do in a code review).


Also, while local dev for us presumably happens on machines with their localtime set to America/Chicago, it's very likely that the build machine will be set to GMT. You should make sure that your time calculations will work correctly there.